### PR TITLE
Don't Merge: Removed randomising the creation of room ids and instead used uuids.

### LINKED ISSUE #281: Remove Convoluted Way to Create Room
### In-Depth Feature Description
In MafiaGame.js, the method createNewRoom() uses a for loop that randomly assigns the new room an id until it finds an id that is unused. This code seems confusing and the logic convoluted. 

### Images for Reference:
image

### Potential Solutions:
- One solution I can think of is to loop through all the IDs and assign the earliest one that is unassigned found. This seems better than doing it randomly and the code would look easier to understand (if written right).
- Another solution would be to set HashSets to store the available and none-available rooms. This makes it more efficient as it would be constant each time we look for an available room, though we would also need to add this logic to releasing a room.

### Feature Approval:
- [x]  Feature approved by: @AngelaLH

### PR DESCRIPTION
**Pull request by milkyway404**
# Description
This PR fixes #281 

Removed randomising the creation of room ids and instead used uuids. Installed UUID.

# Type of change:
* New feature (Non-breaking change that adds functionality)

### CODE REVIEW

File: backend/Events/GameStateEvents/NightStateChangeEvents.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const config = require('../../config.json');
```
Comment (milkyway404) [inline backend/Events/GameStateEvents/NightStateChangeEvents.js]: not part of this PR, ignore (rebase stuff urgh)

File: backend/Events/GameStateEvents/TrialStateChangeEvents.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 const config = require('../../config.json');
```
Comment (milkyway404) [inline backend/Events/GameStateEvents/TrialStateChangeEvents.js]: not part of this PR, ignore (rebase stuff urgh)

File: backend/domain/Enum/PlayerStatus.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/**
```
Comment (milkyway404) [inline backend/domain/Enum/PlayerStatus.js]: not part of this PR, ignore (rebase stuff urgh)

File: backend/domain/Player.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+const PlayerStatus = require('./Enum/PlayerStatus');
+
 /**
```
Comment (milkyway404) [inline backend/domain/Player.js]: not part of this PR, ignore (rebase stuff urgh)

File: backend/domain/Room.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
 const GameStateEnum = require('./Enum/GameStateEnum');
```
Comment (milkyway404) [inline backend/domain/Room.js]: mostly not part of this PR, ignore (rebase stuff urgh)

the last part about uuids is part of this PR.

File: backend/test/integration/GameStartEvents.integration.test.js [hunk 6]
```diff
@@ -28,13 +28,16 @@ describe('GameStartEvents integration tests', () => {
         SocketIOServer.server.close();
     });
 
-    test('integration test start-game 6 players', async (done) => {
```
Comment (milkyway404) [inline backend/test/integration/GameStartEvents.integration.test.js]: not part of this PR, ignore (rebase stuff urgh)

File: backend/test/integration/IntegrationTestHelpers.js [hunk 7]
```diff
@@ -45,7 +45,7 @@ async function connectAndJoin(clientSockets, index, port, lobbyCode) {
  * @returns Promise with resolve returning role of host player
  */
 let hostRole;
-async function startGame(clientSockets) {
```
Comment (milkyway404) [inline backend/test/integration/IntegrationTestHelpers.js]: not part of this PR, ignore (rebase stuff urgh)

File: backend/test/integration/LobbyEvents.integration.test.js [hunk 8]
```diff
@@ -25,17 +25,20 @@ describe('lobby events tests', () => {
         SocketIOServer.server.close();
     });
     /**
-     * This is a integration test for all lobby events.
```
Comment (milkyway404) [inline backend/test/integration/LobbyEvents.integration.test.js]: not part of this PR, ignore (rebase stuff urgh)

File: backend/test/integration/NightTimeVoteEvents.integration.test.js [hunk 9]
```diff
@@ -27,20 +27,21 @@ describe('night time vote event tests', () => {
         SocketIOServer.server.close();
     });
     /**
-     * This is a integration test for night time voting events.
+     * These are integration tests for night time voting events for a 6 and 7 person game.
      * It goes through stariting a game to checking mafia vote
      *
      */
-    test('Night time Vote events', async (done) => {
```
Comment (milkyway404) [inline backend/test/integration/NightTimeVoteEvents.integration.test.js]: not part of this PR, ignore (rebase stuff urgh)

File: backend/test/integration/StateChangeEvents.integration.test.js [hunk 10]
```diff
@@ -30,42 +30,44 @@ afterAll(() => {
 
 /** -----------------TESTS------------------*/
 describe('NightStateChangeEvents integration tests', () => {
-    test('integration test start-night 6 players', async (done) => {
```
Comment (milkyway404) [inline backend/test/integration/StateChangeEvents.integration.test.js]: not part of this PR, ignore (rebase stuff urgh)

**Review by ODE-BOY — APPROVED** [review]
All tests pass, and the code looks clear and concise. Good job! 
Changing to UUIDs I feel is a smart change. While it's not impossible for duplicate UUID's to be created, the probability is infinitesimally small.

File: backend/domain/Room.js [hunk 11]
```diff
@@ -144,12 +145,12 @@ class Room {
     }
 
     /**
-     * Generate a random alphanumeric id
-     * @returns {string} a random alphanumeric id
+     * Generate a univerally unique identifier
+     * @returns {string} a univerally unique identifier
      */
-    // eslint-disable-next-line class-methods-use-this
+    /* eslint class-methods-use-this: ["error", { "exceptMethods": ["getRandomID"] }] */
```
Comment (lhuijser) [inline backend/domain/Room.js:151]: This is mainly just for my understanding, but could you please explain what this line does?

File: backend/domain/Room.js [hunk 12]
```diff
@@ -144,12 +145,12 @@ class Room {
     }
 
     /**
-     * Generate a random alphanumeric id
-     * @returns {string} a random alphanumeric id
+     * Generate a univerally unique identifier
+     * @returns {string} a univerally unique identifier
```
Comment (lhuijser) [inline backend/domain/Room.js:149]: Good description, just a typo of the word universally here, not a big deal but just to let you know! Not urgent

**Review by lhuijser — APPROVED** [review]
Looks great! Thank you for making this change, it makes the code cleaner :) Just two minor comments.

**Comment by milkyway404** [comment]
Closed because this would cause the room ids to be too long for users. Will revert to original solution for now.
